tea.py

```diff
@@ -0,0 +1,674 @@
+import pybamm
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+class Tea(pybamm.BaseModel):
+    """"""
+
+    def __init__(self, parameter_values=None):
+        self.parameter_values = parameter_values
+        if self.parameter_values is None:
+            print(
+                "Warning: Missing parameter values, please supply parameter values to Tea()"
+            )
+        super().__init__()
+
+    def plot_ocvs(self, parameter_values=None):
+        # plot ocv curves, average values against gravimetric capacity of cam and
+        # what if two different parameter sets are used? if, try, except?
+        # in case it is called two times after each other, the second time it will not work
+        # the solution is to call it the second time
+        parameter_values = self.parameter_values
+        try:
+            U_n = parameter_values["Negative electrode OCP [V]"]
+            U_p = parameter_values["Positive electrode OCP [V]"]
+        except KeyError:
+            print("Error: Missing required parameters")
+            return
+
+        x0, x100, y100, y0 = pybamm.lithium_ion.get_min_max_stoichiometries(
+            parameter_values
+        )
+
+        soc = pybamm.linspace(0, 1)  # cell level soc
+        x = x0 + soc * (x100 - x0)
+        y = y0 - soc * (y0 - y100)
+
+        U_n_entries = U_n(x).entries
+        U_p_entries = U_p(y).entries
+
+        plt.clf()
+        plt.figure()
+        plt.plot(soc.entries, U_p_entries - U_n_entries, label="stack")
+        plt.plot(soc.entries, U_p_entries, label="cathode")
+        plt.plot(soc.entries, U_n_entries, label="anode")
+        plt.xlabel("State of charge")
+        plt.ylabel("Open circuit potential (V)")
+        plt.legend()
+        plt.show()
+
+    def stack_energy_densities(self, input_parameter=None, print_values=False):
+        if input_parameter is None:
+            print("Warning: Missing input parameters")
+
+        """if different parameter (e.g. for alloys or conversion-based cam) are of interest, all parameters can be supplied explicitly"""
+
+        # ocv's
+        ne_ocv = self.parameter_values["Negative electrode OCP [V]"]
+        pe_ocv = self.parameter_values["Positive electrode OCP [V]"]
+        x0, x100, y100, y0 = pybamm.lithium_ion.get_min_max_stoichiometries(
+            self.parameter_values
+        )
+        soc = pybamm.linspace(0, 1)  # cell level soc
+        x = x0 + soc * (x100 - x0)
+        y = y0 - soc * (y0 - y100)
+        if input_parameter is None:
+            ne_average_ocv = ne_ocv(x).entries.mean()
+            pe_average_ocv = pe_ocv(y).entries.mean()
+            stack_average_ocv = pe_average_ocv - ne_average_ocv
+        else:
+            try:
```

Comment:
for setting defaults when something isn't an input you can use `get` and `or`. E.g. if you define 
```
def function(inputs=None):
    inputs = inputs or {}
    a = inputs.get("a") or 5
    return a 
```
then `function()` returns 5 and `function({"a": 10})` returns 10